runtime/storage/cache/cache.go

```diff
@@ -0,0 +1,267 @@
+// Package cache provides the ability to define distributed Redis cache clusters
+// and functionality to use them in a fully type-safe manner.
+//
+// For more information see https://encore.dev/docs/develop/caching
+package cache
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+)
+
+// ClusterConfig represents the configuration of cache clusters.
+type ClusterConfig struct {
+	// EvictionPolicy decides how the cache evicts existing keys
+	// to make room for new data when the max memory limit is reached.
+	//
+	// If not specified the cache defaults to AllKeysLRU.
+	EvictionPolicy EvictionPolicy
+}
+
+// An EvictionPolicy describes how the cache evicts keys to make room for new data
+// when the maximum memory limit is reached.
+//
+// See https://redis.io/docs/manual/eviction/#eviction-policies for more information.
+type EvictionPolicy string
+
+// NOTE: These values need to be added to the runtimeconstants package
+// and the parser package for the parser to be aware of them.
+
+const (
+	// AllKeysLRU keeps most recently used keys and removes least recently used (LRU) keys.
+	// This is a good default choice for most cache use cases if you're not sure.
+	AllKeysLRU EvictionPolicy = "allkeys-lru"
+
+	// AllKeysLFU keeps frequently used keys and removes least frequently used (LFU) keys.
+	AllKeysLFU EvictionPolicy = "allkeys-lfu"
+
+	// AllKeysRandom randomly removes keys as needed.
+	AllKeysRandom EvictionPolicy = "allkeys-random"
+
+	// VolatileLRU removes least recently used keys with an expiration set.
+	// It behaves like NoEviction if there are no keys to evict with an expiration set.
+	VolatileLRU EvictionPolicy = "volatile-lru"
+
+	// VolatileLFU removes least frequently used keys with an expiration set.
+	// It behaves like NoEviction if there are no keys to evict with an expiration set.
+	VolatileLFU EvictionPolicy = "volatile-lfu"
+
+	// VolatileTTL removes keys with an expiration set and the shortest time to live (TTL).
+	// It behaves like NoEviction if there are no keys to evict with an expiration set.
+	VolatileTTL EvictionPolicy = "volatile-ttl"
+
+	// VolatileRandom randomly removes keys with an expiration set.
+	// It behaves like NoEviction if there are no keys to evict with an expiration set.
+	VolatileRandom EvictionPolicy = "volatile-random"
+
+	// NoEviction does not evict any keys, and instead returns an error to the client
+	// when the max memory limit is reached.
+	NoEviction EvictionPolicy = "noeviction"
+)
+
+//publicapigen:keep
+type constStr string
```

Comment:
Can we put a doc on here explaining what it is for people. It seems like a read type initially if you're not used to seeing this